frontend/src/utils/search/convertSearchParamsToProperTypes.ts

```diff
@@ -30,12 +32,18 @@ export function convertSearchParamsToProperTypes(
 }
 
 // Helper function to convert query parameters to set
-function paramToSet(param: string | string[] | undefined): Set<string> {
-  if (!param) return new Set();
-  if (Array.isArray(param)) {
-    return new Set(param);
+// and to reset that status params none if status=none is set
+function paramToSet(param: QuerySetParam, type?: string): Set<string> {
+  const defaultedParam =
```

Comment:
ugh, this logic is more complex than I thought - apologies for the problems in my first suggestion. I think this works though???

```suggestion
function paramToSet(param: QuerySetParam, type?: string): Set<string> {\
  if (!param && type === "status") {
    return new Set(["forecasted", "posted"])
  }

  if (!param || (type === "status" && param === SEARCH_NO_STATUS_VALUE)) {
    return new Set();
  }

  if (Array.isArray(param)) {
    return new Set(param);
  }
  return new Set(param.split(","));
}
```

Reply:
What I created can only be described as the twisted love-child of bad design decisions in typescript and javascript and my own addled mind the night before an important election; and the worst monstrosity I've almost ever wrought on a code base.

This looks better, thanks!